api/users/views.py

```diff
@@ -70,14 +71,25 @@ def get_user(self, check_permissions=True):
             else:
                 return self.request.user
 
-        obj = get_object_or_error(User, key, 'user')
+        obj = get_object_or_error(OSFUser, key, 'user')
         if check_permissions:
             # May raise a permission denied
             self.check_object_permissions(self.request, obj)
         return obj
 
 
-class UserList(JSONAPIBaseView, generics.ListAPIView, ODMFilterMixin):
+class UserFilterSet(JSONAPIFilterSet):
```

Comment:
`FilterSets` and filter field classes should probably be separated into a separate file. Let's move this to `api/users/filters.py`.